### PR TITLE
Fix the incompatible with casc

### LINKED ISSUE #75: Please switch to DataBoundConstructor for mandatory and DataBoundSetter for the rest
Configuring this plugin with configuration-as-code YAML fails if not all properties are defined in the YAML which can be fixed by using @DataBoundConstructor for mandatory and @DataBoundSetter for the rest of the properties.

More details here:
https://github.com/jenkinsci/configuration-as-code-plugin/issues/994#issuecomment-523622115

### PR DESCRIPTION
**Pull request by LinuxSuRen**
fix https://github.com/jenkinsci/configuration-as-code-plugin/issues/994

might be able to fix #75

### CODE REVIEW

**Comment by casz** [comment]
Not really a fix more of a temporary patch.

**Comment by casz** [comment]
The fix is to fix up databinding and add JCasC tests

**Comment by LinuxSuRen** [comment]
@casz Yes, tests are needed.

**Comment by casz** [comment]
@LinuxSuRen You ought to update the parent pom 😓 Just go for broke and go with latest since your updating Jenkins version anyhow.

**Comment by LinuxSuRen** [comment]
> @LinuxSuRen You ought to update the parent pom 😓 Just go for broke and go with latest since your updating Jenkins version anyhow.

Sorry, I guest I didn't understand your meaning.

**Comment by casz** [comment]
Please update the parent pom: https://github.com/jenkinsci/oic-auth-plugin/blob/0d40b50a1565216a955a97072094880a172a362f/pom.xml#L6 😓

**Comment by casz** [comment]
SpotBugs is like this at times

File: pom.xml [hunk 1]
```diff
@@ -124,6 +124,13 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <configuration>
+          <excludeFilterFile>src/findbugs/excludesFilter.xml</excludeFilterFile>
```
Comment (casz) [inline pom.xml]: this shouldn't be needed :)

You should rename the folder I believe
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/spotbugs/excludesFilter.xml

Comment (LinuxSuRen) [reply pom.xml]: I didn't know that before. It's cool.

Comment (casz) [reply pom.xml]: I suggest dropping `Add spotbugs exclude filters` :)

**Comment by ashutoshg** [comment]
@LinuxSuRen @casz 
Can this be merged and released please?.
This will be really really useful for us.
Thanks

**Comment by LinuxSuRen** [comment]
Hi @ashutoshg , I'm still waiting for the result of adoption. See also https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/jenkinsci-dev/mOspSMRtQmg/BjHnAnEQBAAJ.

**Comment by yelhouti** [comment]
@ This would be really useful, for use two. A release after merging would be much appreciated :)

**Comment by casz** [comment]
I prefer not to merge dirty fix. I'd rather prefer a proper fix of data bindings.

**Comment by yelhouti** [comment]
@casz What would you call a proper fix of data bindings? I can work on it if you explain a bit more what you want

**Comment by casz** [comment]
https://github.com/jenkinsci/configuration-as-code-plugin/issues/994#issuecomment-523622115

**Comment by LinuxSuRen** [comment]
It looks like that field `automanualconfigure` is mandatory. I agree with @casz comments from above. But I'd prefer to fix that in another PR.

**Comment by danielfbm** [comment]
any updates on this fix? we kind of rely on this for one of our projects

**Comment by LinuxSuRen** [comment]
@casz Do you agree with me that fixing the issues you mentioned as a sperate PR?

**Comment by LinuxSuRen** [comment]
I'm going to merge this. Please create new PRs if there's anything goes wrong.

**Pull request merged**